src/algorithms/AsymmetricForwardBackwardAdjoint.jl

```diff
@@ -0,0 +1,181 @@
+end
+
+################################################################################
+# Utility methods
+maxit(sol::AFBAIterator) = sol.maxit
+
+converged(sol::AFBAIterator, it) = norm(sol.FPR_x)+norm(sol.FPR_y) <= sol.tol
+
+verbose(sol::AFBAIterator) = sol.verbose > 0
+verbose(sol::AFBAIterator, it) = sol.verbose > 0 && (sol.verbose == 2 ? true : (it == 1 || it%sol.verbose_freq == 0))
+
+function display(sol::AFBAIterator)
+    @printf("%6s | %22s | %10s |\n ", "it", "gamma", "fpr")
+    @printf("------|------------------------|------------|\n")
+end
+
+function display(sol::AFBAIterator, it)
+    @printf("%6d | %7.4e, %7.4e | %7.4e |\n", it, sol.gamma1, sol.gamma2, norm(sol.FPR_x)+norm(sol.FPR_y))
+end
+
+function Base.show(io::IO, sol::AFBAIterator)
+    println(io, "Asymmetric Forward-Backward-Adjoint Splitting" )
+    if sol.theta==2
+        println(io, "theta               : $(sol.theta)" )
+    else 
+        println(io, "theta, mu           : $(sol.theta), $(sol.mu)" )
+    end
+    println(io, "fpr                 : $(norm(sol.FPR_x)+norm(sol.FPR_y))")
+    print(  io, "gamma1, gamma2      : $(sol.gamma1), $(sol.gamma2)")
+end
+
+################################################################################
+# Initialization
+
+function initialize(sol::AFBAIterator)
+    return
+end
+
+################################################################################
+# Iteration
+
+function iterate(sol::AFBAIterator{I, R,T}, it::I) where {I, R, T}
+
+    # considering when h \equiv 0
+    if typeof(sol.h)==Zero
+         # perform x-update step
+        gradient!(sol.gradf,sol.f, sol.x)
+        prox!(sol.xbar, sol.g, sol.x .-sol.gamma1*sol.gradf, sol.gamma1)
+        # perform y-update step
+        sol.FPR_x .= sol.xbar .- sol.x   # fix this later
+        sol.FPR_y .= 0   # fix this later
+        sol.x .= sol.x .+ sol.lam .*(sol.FPR_x) 
+    else
+      # perform x-update step
+        gradient!(sol.gradf,sol.f, sol.x)
+        prox!(sol.xbar, sol.g, sol.x .- sol.L'*(sol.gamma1*sol.y).-sol.gamma1*sol.gradf, sol.gamma1)
+        # perform y-update step
+        gradient!(sol.gradl,sol.l, sol.y)   ############fix  # l is basically l^*!! hence Zero() by default 
+        prox!(sol.ybar, Conjugate(sol.h), sol.y .+ sol.L*(sol.gamma2*(sol.theta*sol.xbar.+(1-sol.theta)*sol.x)).-sol.gamma2*sol.gradl, sol.gamma2) 
+        sol.FPR_x .= sol.xbar .- sol.x   # fix this later
+        sol.FPR_y .= sol.ybar .- sol.y   # fix this later
+        sol.x .= sol.x .+ sol.lam .*(sol.FPR_x .- sol.L'*(sol.mu*(2-sol.theta)*sol.gamma1*sol.FPR_y)) 
+        sol.y .= sol.y .+ sol.lam.*(sol.FPR_y .+ sol.L*((1-sol.mu)*(2-sol.theta)*sol.gamma2*sol.FPR_x))
+    end
+    return sol.x
+end
+
+################################################################################
+# Solver interface(s)
+
+function AFBA(x0,y0; kwargs...)
```

Comment:
Another thing that would be nice to have here is direct access to special cases. Very short and incomplete example, but to get the idea: if the Vu-Condat corresponds to having `theta = 2.0`, then

```
function VuCondat(x0, y0; kwargs...)
    sol = AFBAIterator(x0, y0; kwargs..., theta = 2.0)
    (it, point) = run(sol)
    return (it, point, sol)
end
```

(If any other argument other than `theta` needs to be fixed to some value, you can just put it after `kwargs...` and will simply overwrite whatever may have been set in `kwargs...` by the user)

Reply:
The problem is that most of these special cases don't have a name. In fact apart from Vu-Condat all the other case are either extensions of other algorithms or new.

Reply:
I don't agree. Puya has given them some very fancy names in the paper: Algorithm 1, Algorithm 6, Algorithm 54, etc.

Reply:
The special cases that we have so far are the interesting ones and I don't think any one of them has superiority over the others in general. It is dependent on whether we want a parallel implementation or not and basically which algorithm results in a larger stepsize for a given problem (this has to be checked in demos).  Probably the most interesting one is the one with theta=1, mu =1 but we should see. What I mean is that highlighting one algorithm in the implementation over the others does not make much sense to me. 

In any case I think we should also add a generic version of the algorithm with dynamic stepsize and any theta and mu. The issue with that is to figure out a nice stepsize condition. I would do this later.

Reply:
Agreed